bip-csfs.mediawiki

```diff
@@ -0,0 +1,124 @@
+<pre>
+  BIP: TBD
+  Layer: Consensus (soft fork)
+  Title: CHECKSIGFROMSTACK
+  Author: Brandon Black <freedom@reardencode.com>
+  Comments-Summary: No comments yet.
+  Status: Draft
+  Type: Standards Track
+  Created: 2023-12-22
+  License: PD
+</pre>
+
+==Abstract==
+
+This BIP describes two new opcode for the purpose of checking cryptographic
+signatures in bitcoin scripts against data other than bitcoin transactions.
+
+==Summary==
+
+We propose replacing OP_NOP5 (0xb4) in bitcoin script with
+<code>OP_CHECKSIGFROMSTACKVERIFY</code>. When verifying taproot script spends having
```

Comment:
Why split across two extension mechanisms instead of just focusing on the modern path (op success)?

Reply:
Checking signatures from the stack is something that _can_ be supported on all script types (i.e. is compatible with NOP upgrade), and there are use cases where segwit v0 scripts still make the most sense vs. tapscripts.

If supporting legacy and segwitv0 where possible for new opcodes is not something that others value, switching CSFSV to an OP_SUCCESS upgrade would make sense.